### PR TITLE
[bug](fix) fix geo memory leak 

### PR DESCRIPTION
**Pull request by LemonLiTree**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by xiaokang — DISMISSED** [review]
LGTM

**Comment by LemonLiTree** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.25 seconds
 stream load tsv:          431 seconds loaded 74807831229 Bytes, about 165 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          59 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230425045642_clickbench_pr_134524.html

**Comment by LemonLiTree** [comment]
run buildall

**Comment by LemonLiTree** [comment]
run buildall

File: be/src/geo/geo_tobinary.cpp [hunk 1]
```diff
@@ -93,6 +93,7 @@ bool toBinary::writeGeoPolygon(doris::GeoPolygon* polygon, ToBinaryContext* ctx)
     for (int i = 0; i < coordss->list.size(); ++i) {
         writeCoordinateList(*coordss->list[i], true, ctx);
     }
+    delete coordss;
```
Comment (yiguolei) [inline be/src/geo/geo_tobinary.cpp:96]: Not fix it in this way. Should refactor the code in GeoPolygon and return a unique ptr here.
Current code may memory leak if user call return before line 96

**Comment by LemonLiTree** [comment]
run buildall

**Comment by morningman** [comment]
Not for branch-1.2

**Pull request closed without merging**